session/session.go

```diff
@@ -772,20 +777,27 @@ func (s *Session) Close(force bool) ([]perun.ChInfo, error) {
 		}
 	}
 	if len(unexpectedPhaseChIDs) != 0 {
-		err := fmt.Errorf("chs in unexpected phase during session close: %v", unexpectedPhaseChIDs)
-		s.Error(err.Error())
+		err := ErrunexpectedPhaseChIDs
+		apiErr = perun.NewAPIErrV2FailedPreCondition(err.Error())
+		s.WithFields(perun.APIErrV2AsMap("Close", apiErr)).Error(apiErr.Message())
 		s.unlockAllChs()
-		return nil, perun.GetAPIError(errors.WithStack(err))
+		return nil, apiErr
 	}
 	if !force && len(openChsInfo) != 0 {
-		err := fmt.Errorf("%w: %v", perun.ErrOpenCh, openChsInfo)
-		s.Error(err.Error())
+		apiErr = perun.NewAPIErrV2FailedPreCondition(ErrOpenCh.Error())
```

Comment:
Can you update the `NewAPIErrV2FailedPreconditio` to take `addInfo` as an argument. And in this case, you can create an addInfo type with the a field containing open channels info ?